spec/features/consumer/shopping/cart_spec.rb

```diff
@@ -56,9 +56,14 @@
       let(:variant) { product_tax.variants.first }
 
       before do
+        Spree::Config.set allow_backorders: false
         add_product_to_cart order, product_tax
       end
 
+      after do
+        Spree::Config.set allow_backorders: true
```

Comment:
Ok, so the reason that stubbing `:allow_backorders` doesn't work is that there are two way of accessing these properties 1) `Spree::Config.allow_backorders` and `Spree::Config[:allow_backorders]`. The second way is not going to be affected by the stub, because it's a different method call (`[]`). The code that this particular spec is testing is in `app/overrides/spree/products/_cart_form/only_show_current_distribution_variants.html.haml.deface` and it uses the hash method to access the setting, so that is why it is not working.